tests/self/Basic/080AdvectionLinear_SelfTest.cpp

```diff
@@ -191,58 +222,80 @@ class AdvectionLinear : public Base::HpgemAPILinear<DIM> {
-        } else if (dim[i] == 2) {
-            AdvectionLinear<2> test(fileName, p[i]);
-            error = test.createAndSolve(T[i], nT[i]);
-            logger(INFO, "Error: % (expected %)", error, errors[i]);
-            logger.assert_always((std::abs(error - errors[i]) < 1e-8),
-                                 "comparison to old results");
-        } else {
-            AdvectionLinear<3> test(fileName, p[i]);
-            error = test.createAndSolve(T[i], nT[i]);
-            logger(INFO, "Error: % (expected %)", error, errors[i]);
-            // logger.assert_always((std::abs(error - errors[i]) < 1e-8),
-            // "comparison to old results");
-        }
-    }
+
+    // For recomputing the error tables
+    bool ignoreFailures = false;
+
+    AdvectionParameters dim1Params = {1, 0.1, 10};
+    // Minimum of 4 elements
+    ConvergenceTestSet dim1Meshes = {getUnitSegmentMeshes(2),
+                                     {
+                                         6.59544434e-02,  //------
+                                         1.79880373e-02,  //  3.67
+                                         5.18317146e-03,  //  3.47
+                                         1.51017550e-03,  //  3.43
+                                     }};
+
+    runAdvectionTestSet<1>(dim1Meshes, dim1Params, ignoreFailures);
+
+    ConvergenceTestSet dim2P1Meshes = {getUnitSquareTriangleMeshes(2),
+                                       {
+                                           1.03268522e-01,  //------
+                                           2.27752735e-02,  //  4.53
+                                           6.72435135e-03,  //  3.39
+                                           2.00067031e-03,  //  3.36
+                                           5.52269096e-04,  //  3.62
+                                       }};
+    AdvectionParameters dim2P1Params = {1, 0.1, 10};
+    runAdvectionTestSet<2>(dim2P1Meshes, dim2P1Params, ignoreFailures);
+
+    // Second order dim2
+    ConvergenceTestSet dim2P2Meshes = {getUnitSquareTriangleMeshes(1, 5),
+                                       {
+                                           6.64556121e-02,  //------
+                                           7.11147068e-03,  //  9.34
+                                           2.71258393e-03,  //  2.62
+                                           3.91992217e-04,  //  6.92
+                                       }};
+    AdvectionParameters dim2P2Params = {2, 0.1, 10};
+    runAdvectionTestSet<2>(dim2P2Meshes, dim2P2Params, ignoreFailures);
+
+    // First order dim 3
+    ConvergenceTestSet dim3Meshes = {getUnitCubeCubeMeshes(2, 5),
+                                     {
+                                         5.55152172e-02,  //------
+                                         1.43980436e-02,  //  3.86
+                                         3.81712900e-03,  //  3.77
+                                     }};
+    AdvectionParameters dim3Params = {1, 0.02, 2};
```

Comment:
What/where specifically, because every file is now documented in what numerical problem it solves and what convergence rate to expect. The tables as you see them here are rather standard in FEM literature (though there with less precision).

I did add a line in ConvergenceTest to hint at its intended usage.

Reply:
thanks